hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -1775,7 +1775,7 @@ public OzoneKeyDetails getS3KeyDetails(String bucketName, String keyName,
           .filter(omKeyLocationInfo -> omKeyLocationInfo.getPartNumber() ==
                                        partNumber)
           .collect(Collectors.toList());
-      keyInfo.updateLocationInfoList(filteredKeyLocationInfo, false);
+      keyInfo.updateLocationInfoList(filteredKeyLocationInfo, false, true);
```

Comment:
Nit: Can use `keyInfo.updateLocationInfoList(filteredKeyLocationInfo, true, true)`, although it doesn't really metter.

Reply:
Thanks, done.